message.go

```diff
@@ -10,10 +10,13 @@ import (
 
 // Message is a data structure representing kafka messages.
 type Message struct {
-	// Topic is reads only and MUST NOT be set when writing messages
+	// Topic may be set to specify the topic for this message, but only if the
```

Comment:
Maybe add documentation as well to explain that for messages returned by a reader, the field is set to the topic that the message was read from?

Reply:
Good callout!